modules/siddhi-extensions/output-mappers/json-output-mapper/src/main/java/org/wso2/siddhi/extension/output/mapper/json/JSONOutputMapper.java

```diff
@@ -71,6 +80,7 @@ public void mapAndSend(Event[] events, OutputTransportCallback outputTransportCa
                 outputTransportCallback.publish(payloadTemplateBuilder.build(event), event);
             }
         } else {
+            //TODO add support to publish multiple events
             for (Event event : events) {
```

Comment:
Since events is an array, it will first have to be converted to an iterable. It is uncertain whether there'll be much of a performance improvement by converting this to a Java8 forEach.